mmv1/templates/terraform/encoders/cloud_run_domain_mapping.go.erb

```diff
@@ -13,8 +13,13 @@
 	# limitations under the License.
 -%>
 name := d.Get("name").(string)
-metadata := obj["metadata"].(map[string]interface{})
+raw := obj["metadata"]
```

Comment:
`metadata` is required and not computed in this resource and `setNew` cannot be applied to it. So I change it to `default_from_api: true` to apply the `setNew` method.

With this change, if metadata is not provided by the user, it needs this check and transformation.

Reply:
Are we able to call SetNew on `metadata.0.terraform_labels` / `metadata.0.effective_labels` directly rather than on `metadata` itself? That would let us preserve the existing schema.

Reply:
That was my original thought. bit it is not working. It looks like a known issue in Terraform core. https://github.com/hashicorp/terraform-plugin-sdk/issues/459

Reply:
Darn. Thanks!

We should be able to check if `metadata.#` has a value of 1 in a CustomizeDiff prior to calling SetNew, and return an error if it's not `1`. That'll simulate it being required. https://github.com/hashicorp/terraform-provider-google-beta/blob/2b630c8cddd328ffa55a0e7403854df45d2b3dbf/google-beta/services/bigtable/resource_bigtable_instance.go#L565-L571 was a similar check in BQ.

Reply:
The CustomizeDiff function is shared by other resources, like `google_cloud_run_service`.  But metadata is not required. I can create a new CustomizeDiff function for this resource `google_cloud_run_domain_mapping` to check if metadata is required.